examples/server/utils.hpp

```diff
@@ -122,8 +122,26 @@ inline std::string format_chat(const struct llama_model * model, const std::stri
 
     for (size_t i = 0; i < messages.size(); ++i) {
         const auto & curr_msg = messages[i];
-        std::string role    = json_value(curr_msg, "role",    std::string(""));
-        std::string content = json_value(curr_msg, "content", std::string(""));
+
+        std::string role = json_value(curr_msg, "role", std::string(""));
+
+        std::string content;
+        if (curr_msg.contains("content")) {
+            if (curr_msg["content"].is_string()) {
+                content = curr_msg["content"].get<std::string>();
+            } else if (curr_msg["content"].is_array()) {
+                for (const auto & part : curr_msg["content"]) {
+                    if (part.contains("text")) {
+                        content += part["text"].get<std::string>();
```

Comment:
I haven't tested, but should we put a new line to separate between multiple texts in a message?

```suggestion
                        content += "\n" + part["text"].get<std::string>();
```

Reply:
Yes, I guess it makes sense, though it's not clear from the documentation if that is the required behavior

Reply:
IMO I think the array of `content` would make more sense if used with image, for example user can send an array of `[text, image, text, image]`

But we don't care about image for now, so I think it's not very important how we concatenate multiple texts. My idea is mostly to "make it work" for now. Probably it will be more clear when we bring back llava to server.